Docs: add documentation for new stacking option in Time series panel (#33023)
grafana/grafana: docs/sources/panels/visualizations/time-series/graph-time-series-stacking.md

+++
title = "Graph stacked time series"
keywords = ["grafana", "time series panel", "documentation", "guide", "graph"]
weight = 400
+++

# Graph stacked time series

> > **Note:** This is a beta feature. Time series panel is going to replace the Graph panel in the future releases.

Reviewer: ```suggestion
> **Note:** This is a beta feature. Time series panel is going to replace the Graph panel in a future release.
```

> This section explains how to use Time series field options to control the stacking of the series and illustrates what the stacking options do.

Reviewer: ```suggestion
This section explains how to use Time series panel field options to control the stacking of the series and illustrates what the stacking options do.
```

Reviewer: Can you explain a bit more about what stacking is and why someone might want to do it?

Reviewer 2: can i write a followup section about how you shouldn't do it? ;)

https://leeoniya.github.io/uPlot/demos/stacked-series.html

Reviewer: If you like, sure. You could also write an educational topic about how to create graphs, what types of graphs go with what data, and that sort of thing. Our long-term goal is to include more basic information for beginners, so that would be a big help!

Author: @leeoniya I think we should just copy the paragraph from your demo, I always enjoy reading it 🙈

Use the following field settings to configure your series stacking.

For more information about applying these options, refer to:

- [Configure all fields]({{< relref "../../field-options/configure-all-fields.md" >}})
- [Configure specific fields]({{< relref "../../field-options/configure-specific-fields.md" >}})

Some field options will not affect the visualization until you click outside of the field option box you are editing or press Enter.

## Stack series

![Stack series editor](/img/docs/time-series-panel/stack-series-editor-8-0.png)

> Enable/disable series stacking

Reviewer: Put the text above the image and make it a complete sentence.

Reviewer: This applies to all entries.

### Off

Do not stack series

![No series stacking example](/img/docs/time-series-panel/stacking-off-8-0.png)

### Normal

Stack series on top of each other

![Normal series stacking example](/img/docs/time-series-panel/stacking-normal-8-0.png)

## Stack series in groups

> **Note:** Stacking groups are available via the field overrides

For more information about creating field overrides, refer to [Add a field override]({{< relref "../../field-options/configure-specific-fields.md#add-a-field-override" >}}) 

The following image shows a graph with two stacking groups (A & B) defined.

![Stacking groups example](/img/docs/time-series-panel/stack-series-groups-8-0.png)

> 1. Create a field override for **Stack series** option.

Reviewer: Try to match what I did in this section: https://grafana.com/docs/grafana/latest/panels/visualizations/time-series/graph-time-series-as-lines/#fill-below-to

+++
title = "Graph stacked time series"
keywords = ["grafana", "time series panel", "documentation", "guide", "graph"]
weight = 400
+++

# Graph stacked time series

> **Note:** This is a beta feature. Time series panel is going to replace the Graph panel in a future release.

This section explains how to use Time series panel field options to control the stacking of the series and illustrates what the stacking options do.

> Stacking allows displaying series on top of each other. Be cautious when using stacking in the visualization as it can easily create misleading graphs. You can read more on why stacking may be not the best approach here: [Stacked Area Graphs Are Not Your Friend](https://everydayanalytics.ca/2014/08/stacked-area-graphs-are-not-your-friend.html) 

Reviewer: ```suggestion
_Stacking_ allows Grafana to display series on top of each other. Be cautious when using stacking in the visualization as it can easily create misleading graphs. You can read more on why stacking may be not the best approach here: [Stacked Area Graphs Are Not Your Friend](https://everydayanalytics.ca/2014/08/stacked-area-graphs-are-not-your-friend.html).
```

Reviewer: Does stacking only apply to line graphs, or can you do it with bars and dots as well? If so, I recommend adding at least one non-line example.

Reviewer: @dprokop ^

> Stacking allows displaying series on top of each other. Be cautious when using stacking in the visualization as it can easily create misleading graphs. You can read more on why stacking may be not the best approach here:  [Stacked Area Graphs Are Not Your Friend](https://everydayanalytics.ca/2014/08/stacked-area-graphs-are-not-your-friend.html) 

Author: ```suggestion
Stacking allows displaying series on top of each other. Be cautious when using stacking in the visualization as it can easily create misleading graphs. You can read more on why stacking may be not the best approach here: [Stacked Area Graphs Are Not Your Friend](https://everydayanalytics.ca/2014/08/stacked-area-graphs-are-not-your-friend.html) 
```

Use the following field settings to configure your series stacking.

For more information about applying these options, refer to:

- [Configure all fields]({{< relref "../../field-options/configure-all-fields.md" >}})
- [Configure specific fields]({{< relref "../../field-options/configure-specific-fields.md" >}})

Some field options will not affect the visualization until you click outside of the field option box you are editing or press Enter.

## Stack series

> Turn series stacking on/off.

Reviewer: ```suggestion
Turn series stacking on or off.
```

![Stack series editor](/img/docs/time-series-panel/stack-series-editor-8-0.png)

### Off

> Disable series stacking.

Reviewer: ```suggestion
Turn off series stacking. A series will share the same space in the visualization.
```

![No series stacking example](/img/docs/time-series-panel/stacking-off-8-0.png)

### Normal

Enable stacking series on top of each other.

![Normal series stacking example](/img/docs/time-series-panel/stacking-normal-8-0.png)

## Stack series in groups

> Stacking group option is only available as an override.

Reviewer: ```suggestion
The stacking group option is only available as an override.
```

Reviewer: Why are we calling it "stacking group" if those words do not appear in the UI?

Author: They do, as a placeholder in the stacking group input

For more information about creating field overrides, refer to [Add a field override]({{< relref "../../field-options/configure-specific-fields.md#add-a-field-override" >}}) 

Stack series in the same group. In the Overrides section:

1. Create a field override for **Stack series** option.

![Stack series override](/img/docs/time-series-panel/stacking-override-default-8-0.png)

1. Click on **Normal** stacking mode.
   
1. Name the stacking group you want the series to appear in. The stacking group name option is only available when creating an override.

![Stack series override](/img/docs/time-series-panel/stack-series-override-editor-8-0)

> A-series and B-series stacked in group A, C-series and D-series stacked in group B:

Reviewer: ```suggestion
A-series and B-series stacked in group A, C-series, and D-series stacked in group B:
```
